src/main.cpp

```diff
@@ -859,22 +890,51 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
         CTxMemPoolEntry entry(tx, nFees, GetTime(), dPriority, chainActive.Height(), mempool.HasNoInputsOf(tx));
         unsigned int nSize = entry.GetTxSize();
 
-        // Don't accept it if it can't get into a block
-        CAmount txMinFee = GetMinRelayFee(tx, nSize, true);
-        if (fLimitFree && nFees < txMinFee)
-            return state.DoS(0, error("AcceptToMemoryPool: not enough fees %s, %d < %d",
-                                      hash.ToString(), nFees, txMinFee),
-                             REJECT_INSUFFICIENTFEE, "insufficient fee");
-
-        // Require that free transactions have sufficient priority to be mined in the next block.
-        if (GetBoolArg("-relaypriority", true) && nFees < ::minRelayTxFee.GetFee(nSize) && !AllowFree(view.GetPriority(tx, chainActive.Height() + 1))) {
-            return state.DoS(0, false, REJECT_INSUFFICIENTFEE, "insufficient priority");
+        // The fees required to accept this transaction start with the fees required to accept it on its own
```

Comment:
Can't all this new code be part of  mempool.StageTrimToSize ?
In fact, I would go further and take anything that depends on nFeesDeleted to txmempool as well (including ```csFreeLimiter``` ```GetBoolArg("-relaypriority", true)``` and ```GetMinRelayFee```).
I would also move https://github.com/bitcoin/bitcoin/blob/master/src/main.cpp#L785 there and have all mempool-replacement-related things in one place: txmempool. We could also move some things directly to policy/fees but I'm not going to ask for that much.

I'm sorry but I won't push my nits as branches with fixup! commits anymore on this subject.
I guess I'll have to wait the ages that will take to move the code later and accept that nobody else but me cares about  the movement being better-than-free right now. I won't miss the chance to beg you though.
Please, try some of my movement suggestions and if takes you too long to adapt the parameter too recompile or you don't like what my suggestions are doing to the number of parameters in pool.StageTrimToSize just stop it. And I guess I can just add/maintain all those things in my tomove list. It seems I'm the only one that doesn't love the fact that most of our changes conflict with each other in main.cpp. But I understand, moving things out of main so that things don't need to conflict and don't need to be prioritized is not a priority.
@sipa I must be getting something wrong or I've been communicating poorly around this. Maybe code doesn't talk as much as I thought after all.